packages/walletconnect-v2/src/index.ts

```diff
@@ -48,27 +61,30 @@ export class WalletConnect extends Connector {
   public provider?: WalletConnectProvider
   public readonly events = new EventEmitter3()
 
-  private readonly options: Omit<WalletConnectOptions, 'rpcMap' | 'chains'>
-
-  private readonly rpcMap?: Record<number, string | string[]>
-  private readonly chains: number[]
   private readonly defaultChainId?: number
-  private readonly timeout: number
-
   private eagerConnection?: Promise<WalletConnectProvider>
+  private readonly options: Omit<WalletConnectOptions, 'rpcMap'>
+  private readonly rpcMap?: Record<number, string | string[]>
+  private readonly timeout: number
 
-  constructor({ actions, options, defaultChainId, timeout = DEFAULT_TIMEOUT, onError }: WalletConnectConstructorArgs) {
+  constructor({ actions, defaultChainId, options, timeout = DEFAULT_TIMEOUT, onError }: WalletConnectConstructorArgs) {
     super(actions, onError)
 
-    const { rpcMap, rpc, chains, ...rest } = options
-
-    this.options = rest
-    this.chains = chains
```

Comment:
Also, I believe that, once this line of code is back, you should put the if statement here to check for number of chains (and optional chains) to ensure initialisation is valid and throw an error during the runtime.

Note that this can't be changed once created, so doing if condition here makes definitely more sense instead of at the lazy init time.

Reply:
I would also recommend putting `optionalChains` outside so that we can perform necessary reordering/check for presence.

Reply:
this makes sense given the approach you outlined above